core-parser/src/test/java/org/uniprot/core/parser/fasta/uniprot/UniProtKBFastaParserTest.java

```diff
@@ -238,6 +237,27 @@ void testToUniProtKBFasta(){
         assertEquals(entry.getProteinExistence(), result.getProteinExistence());
         assertEquals(entry.getEntryAudit().getSequenceVersion(), result.getSequenceVersion());
         assertEquals(entry.getSequence(), result.getSequence());
+        assertNull(result.getSequenceRange());
+    }
+
+    @Test
+    void testToUniProtKBFastaWithSubsequenceRange(){
+        List<Gene> genes = List.of(new GeneBuilder().orfNamesAdd(new ORFNameBuilder().value("ORF 1").build()).build());
+        UniProtKBEntry entry = getEntryWithGene(genes);
+        String sequenceRange = "10-20";
+        UniProtKBFasta result = UniProtKBFastaParser.toUniProtKBFasta(entry, sequenceRange);
+        assertNotNull(result);
+        assertEquals(entry.getPrimaryAccession().getValue(), result.getId());
+        assertEquals(entry.getEntryType(), result.getEntryType());
+        assertEquals(entry.getUniProtkbId(), result.getUniProtkbId());
+        assertEquals("Rec Name Value", result.getProteinName());
+        assertEquals("ORF 1", result.getGeneName());
+        assertEquals(entry.getOrganism(), result.getOrganism());
+        assertEquals("FRAGMENTS_PRECURSOR", result.getFlagType().name());
+        assertEquals(entry.getProteinExistence(), result.getProteinExistence());
+        assertEquals(entry.getEntryAudit().getSequenceVersion(), result.getSequenceVersion());
+        assertEquals(entry.getSequence().getValue().substring(9, 20), result.getSequence().getValue());
+        assertEquals(sequenceRange, result.getSequenceRange());
     }
```

Comment:
Can you please add test for subsequence out of range? 
For example, in a sequence with length 50, the user requests a subsequence [60-90] both out of range or [40-80] just end out of range?